### DISCUSSION TITLE
HTTPS via rocket is not recommended

### DISCUSSION THREAD

**Discussion by markohrastovec** [dani-garcia/vaultwarden#3242] [Q&A] [answered]
Why is HTTPS via rocket not recommended according to https://github.com/dani-garcia/vaultwarden/wiki/Enabling-HTTPS?

I already have another docker with HTTPS and published port from docker (HomeAssistant). Now, I would like to add vaultwarden. They will both be hosted on a nanopi. If I forward both ports through ISP routes, they can both use the same SSL certificates, and I do not need any reverse proxy. I would rather not install a web server for reverse proxy, because I do not know how HomeAssistant would behave over reverse proxy, and it requires less work just to install another docker server, and publish its port.

On the other hand, using one reverse proxy server fot both docker images, already make things a bit easier, because I have to maintain SSL certs in one place only.

As I said I am primarily interested, why is HTTPS via rocket not recommended. Based on that information, I will decide whether to install a reverse proxy for all my docker servers or leave every onw with individual HTTPS/SSL configuration.

**Comment by BlackDex** [comment, marked as answer]
Mainly it is the web-sockets. Those will not work without a reverse proxy, since we aren't able currently to run that through rocket on the same port.

Also, while hyper (the http server used by rocket) is getting better, in my opinion, most reverse proxy are better suited for a wide range of protocols, because those are around longer and have a larger user base.

So, without using web-sockets, it should work well i think. And using modern clients should not be an issue.

**Comment by markohrastovec** [comment]
Thanks. Since I am not sure what clients are going to be used, I will probably setup a proper reverse proxy.
